src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityFluidHatch.java

```diff
@@ -212,93 +218,84 @@ public void registerAbilities(List<IFluidTank> abilityList) {
-
-        // Add general widgets
-        return builder.label(6, 6, title)
-                .label(11, 20, "gregtech.gui.fluid_amount", 0xFFFFFF)
-                .widget(new AdvancedTextWidget(11, 30, getFluidAmountText(tankWidget), 0xFFFFFF))
-                .widget(new AdvancedTextWidget(11, 40, getFluidNameText(tankWidget), 0xFFFFFF))
-                .widget(tankWidget)
-                .widget(new FluidContainerSlotWidget(importItems, 0, 90, 16, false)
-                        .setBackgroundTexture(GuiTextures.SLOT, GuiTextures.IN_SLOT_OVERLAY))
-                .bindPlayerInventory(entityPlayer.inventory);
-    }
-
-    protected Consumer<List<ITextComponent>> getFluidNameText(TankWidget tankWidget) {
-        return (list) -> {
-            TextComponentTranslation translation = tankWidget.getFluidTextComponent();
-            // If there is no fluid in the tank, but there is a locked fluid
-            if (translation == null) {
-                translation = GTUtility.getFluidTranslation(this.lockedFluid);
-            }
-
-            if (translation != null) {
-                list.add(translation);
-            }
-        };
-    }
-
-    protected Consumer<List<ITextComponent>> getFluidAmountText(TankWidget tankWidget) {
-        return (list) -> {
-            String fluidAmount = "";
-
-            // Nothing in the tank
-            if (tankWidget.getFormattedFluidAmount().equals("0")) {
-                // Display Zero to show information about the locked fluid
-                if (this.lockedFluid != null) {
-                    fluidAmount = "0";
-                }
-            } else {
-                fluidAmount = tankWidget.getFormattedFluidAmount();
-            }
-            if (!fluidAmount.isEmpty()) {
-                list.add(new TextComponentString(fluidAmount));
-            }
-        };
+    @Override
+    public ModularPanel buildUI(PosGuiData guiData, PanelSyncManager guiSyncManager) {
+        var fluidSyncHandler = GTFluidSlot.sync(fluidTank)
+                .showAmount(false)
+                .accessibility(true, false)
+                .onLockFluid(() -> this.lockedFluid, stack -> {
+                    if (!isExportHatch) return;
+                    setLocked(stack);
+                });
+
+        return GTGuis.createPanel(this, 176, 166)
+                .child(IKey.lang(getMetaFullName()).asWidget().pos(6, 6))
+
+                // export specific
+                .childIf(isExportHatch, new ItemSlot()
+                        .pos(90, 44)
+                        .background(GTGuiTextures.SLOT, GTGuiTextures.OUT_SLOT_OVERLAY)
+                        .slot(new ModularSlot(exportItems, 0)
+                                .accessibility(false, true)))
+                .childIf(isExportHatch, new ToggleButton()
+                        .pos(7, 63)
+                        .overlay(GTGuiTextures.BUTTON_LOCK)
+                        // todo doing things this way causes flickering if it fails
```

Comment:
Why not instead of flickering when there is no fluid stack in the output hatch, go ahead and lock the fluid hatch, and then update the locked fluid the first time a new fluid enters the hatch?

This is a bit of a behavior change, but it could be used to prevent the flickering seen here when clicking the button.